nemo_curator/modules/fuzzy_dedup.py

```diff
@@ -1651,52 +1651,83 @@ def _write_encoded_jaccard_pair(self, dedup_parsed_id_path):
                 )
                 id_columns = ["dataset_id", "doc_id"]
 
-            num_workers = get_num_workers(get_current_client())
-            self._batched_merge_and_write(
+            self._merge_and_write(
                 ddf=ddf,
                 ddf_id=ddf_id,
                 output_path=output_path,
                 id_columns=id_columns,
-                batch_size=num_workers,
             )
         self._logger.info(
             f"Time taken for Encoding Jaccard Pairs = {time.time() - t0}s and output written at {output_path}"
         )
         return output_path
 
-    def _batched_merge_and_write(
-        self, ddf, ddf_id, output_path, id_columns, batch_size=32
-    ):
-        total_batches = (ddf.npartitions + batch_size - 1) // batch_size
-        for batch_id, offset in enumerate(range(0, ddf.npartitions, batch_size)):
-            st = time.time()
-            subset_ddf = ddf.partitions[offset : offset + batch_size]
+    def _merge_and_write(
+        self,
+        ddf: dask_cudf.DataFrame,
+        ddf_id: dask_cudf.DataFrame,
+        output_path: str,
+        id_columns: Union[str, List[str]],
+    ) -> None:
+        st = time.time()
+        # Ensure 'id_columns' is a list
+        if isinstance(id_columns, str):
+            id_columns = [id_columns]
+
+        # Check if there are multiple id_columns
+        if len(id_columns) == 1:
```

Comment:
@ayushdg , Do we have use cases of `convert_str_ids = True` or we can just nuke it ?  

I have added support for now but wonder if we should just remove it .

Reply:
Removed so resolved